position_in_partition.hh

```diff
@@ -676,3 +676,41 @@ bool position_range::overlaps(const schema& s, position_in_partition_view start,
     position_in_partition::less_compare less(s);
     return !less(end, _start) && less(start, _end);
 }
+
+// Track position_in_partittion transitions
+// and validate monotonicity.
+class position_in_partition_tracker {
+    bool _in_stream = false;
+    partition_region _pos = partition_region::partition_start;
+public:
+    void on_partition_start();
+    void on_static_row();
+    void on_clustering_row();
+    void on_partition_end();
+    void on_end_of_stream();
+
+    bool in_stream() {
```

Comment:
const

The same for the other accessors below.

Reply:
Right.  Thanks.

Reply:
Dropped in v2.